### ISSUE TITLE
rapidjson internal assertion failure: IsObject()

### ISSUE THREAD

**Issue by sunnyyakima** [COMBINE-lab/salmon#251]
OS: ubuntu 16.04
Salmon was installed using conda

Version Info: This is the most recent version of Salmon.
### [ output ] => { /scratch/yu_liu/HS_1-1_quant }
Logs will be written to /scratch/yu_liu/HS_1-1_quant/logs
[2018-07-13 20:04:48.086] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.
[2018-07-13 20:04:48.086] [jointLog] [info] parsing read library format
[2018-07-13 20:04:48.086] [jointLog] [info] There is 1 library.
Exception : [rapidjson internal assertion failure: IsObject()]
salmon quant was invoked improperly.
For usage information, try salmon quant --help
Exiting.

**Comment by rob-p** [comment]
Hi @sunnyyakima,

  Can you please list the contents of `/work/yu_liu/resource/salmon_gencodev28_index/headerInfo.json`?  This seems like the type of issue that my arise from not re-building the index (salmon 0.10 has additions to the index from previous versions, and so all indices need to be rebuilt).

**Comment by sunnyyakima** [comment]
all json files are empty files
When I did the index, it seems there is no error.
ls -lrth *json
-rw-r--r-- 1 yu_liu data-sci ==> 0 <== Jul 13 19:47 header.json
-rw-r--r-- 1 yu_liu data-sci ==>0 <==  Jul 13 19:47 refInfo.json
-rw-r--r-- 1 yu_liu data-sci  ==>0 <== Jul 13 19:47 versionInfo.json
(base)

**Comment by rob-p** [comment]
I see.  Well, this is definitely what is leading to the error.  The JSON files shouldn't be empty.  Can you report the output of `salmon quant` when you were building the index?  It seems that the JSON files were not written correctly, which prevents proper loading of the index (and which may be indicative of other parts of the index not being properly written).

**Comment by sunnyyakima** [comment]
just rebuild the index, it seems OK

salmon index -t /work/yu_liu/resource/salmon_reference/gencode.v28.transcripts.fa  -i salmon_gencodev28_index
Version Info: This is the most recent version of Salmon.
index ["salmon_gencodev28_index"] did not previously exist  . . . creating it
[2018-07-13 21:27:37.482] [jLog] [info] building index
RapMap Indexer

[Step 1 of 4] : counting k-mers
[2018-07-13 21:27:42.500] [jointLog] [warning] Entry with header [ENST00000473810.1|ENSG00000239255.1|OTTHUMG00000157482.1|OTTHUMT00000348942.1|RP11-145M9.2-001|RP11-145M9.2|25|processed_pseudogene|], had length less than the k-mer length of 31 (perhaps after poly-A clipping)
[2018-07-13 21:27:42.718] [jointLog] [warning] Entry with header [ENST00000603775.1|ENSG00000271544.1|OTTHUMG00000184300.1|OTTHUMT00000468575.1|AC006499.9-001|AC006499.9|23|processed_pseudogene|], had length less than the k-mer length of 31 (perhaps after poly-A clipping)
[2018-07-13 21:27:46.047] [jointLog] [warning] Entry with header [ENST00000632684.1|ENSG00000282431.1|OTTHUMG00000190602.2|OTTHUMT00000485301.2|RP11-520H11.10-001|TRBD1|12|TR_D_gene|], had length less than the k-mer length of 31 (perhaps after poly-A clipping)
[2018-07-13 21:27:50.627] [jointLog] [warning] Entry with header [ENST00000626826.1|ENSG00000281344.1|OTTHUMG00000189570.1|OTTHUMT00000479989.1|RP11-210L7.2-001|HELLPAR|205012|macro_lncRNA|] was longer than 200000 nucleotides.  Are you certain that we are indexing a transcriptome and not a genome?
[2018-07-13 21:27:50.744] [jointLog] [warning] Entry with header [ENST00000543745.1|ENSG00000255972.1|OTTHUMG00000168883.1|OTTHUMT00000401485.1|RP11-324E6.8-001|RP11-324E6.8|28|processed_pseudogene|], had length less than the k-mer length of 31 (perhaps after poly-A clipping)
[2018-07-13 21:27:51.076] [jointLog] [warning] Entry with header [ENST00000415118.1|ENSG00000223997.1|OTTHUMG00000170844.2|OTTHUMT00000410670.2|AE000661.52-001|TRDD1|8|TR_D_gene|], had length less than the k-mer length of 31 (perhaps after poly-A clipping)
[2018-07-13 21:27:51.076] [jointLog] [warning] Entry with header [ENST00000434970.2|ENSG00000237235.2|OTTHUMG00000170845.2|OTTHUMT00000410671.2|AE000661.53-001|TRDD2|9|TR_D_gene|], had length less than the k-mer length of 31 (perhaps after poly-A clipping)
[2018-07-13 21:27:51.076] [jointLog] [warning] Entry with header [ENST00000448914.1|ENSG00000228985.1|OTTHUMG00000170846.2|OTTHUMT00000410672.2|AE000661.54-001|TRDD3|13|TR_D_gene|], had length less than the k-mer length of 31 (perhaps after poly-A clipping)
[2018-07-13 21:27:54.007] [jointLog] [warning] Entry with header [ENST00000439842.1|ENSG00000236597.1|OTTHUMG00000152435.2|OTTHUMT00000326213.2|AL122127.38-001|IGHD7-27|11|IG_D_gene|], had length less than the k-mer length of 31 (perhaps after poly-A clipping)
[2018-07-13 21:27:54.007] [jointLog] [warning] Entry with header [ENST00000390567.1|ENSG00000211907.1|OTTHUMG00000152429.2|OTTHUMT00000326207.2|AL122127.37-001|IGHD1-26|20|IG_D_gene|], had length less than the k-mer length of 31 (perhaps after poly-A clipping)
[2018-07-13 21:27:54.007] [jointLog] [warning] Entry with header [ENST00000452198.1|ENSG00000225825.1|OTTHUMG00000152436.2|OTTHUMT00000326214.2|AL122127.36-001|IGHD6-25|18|IG_D_gene|], had length less than the k-mer length of 31 (perhaps after poly-A clipping)
[2018-07-13 21:27:54.007] [jointLog] [warning] Entry with header [ENST00000390569.1|ENSG00000211909.1|OTTHUMG00000152427.2|OTTHUMT00000326205.2|AL122127.35-001|IGHD5-24|20|IG_D_gene|], had length less than the k-mer length of 31 (perhaps after poly-A clipping)
[2018-07-13 21:27:54.007] [jointLog] [warning] Entry with header [ENST00000437320.1|ENSG00000227196.1|OTTHUMG00000152438.2|OTTHUMT00000326216.2|AL122127.34-001|IGHD4-23|19|IG_D_gene|], had length less than the k-mer length of 31 (perhaps after poly-A clipping)
[2018-07-13 21:27:54.007] [jointLog] [warning] Entry with header [ENST00000390572.1|ENSG00000211912.1|OTTHUMG00000152428.2|OTTHUMT00000326206.2|AL122127.32-001|IGHD2-21|28|IG_D_gene|], had length less than the k-mer length of 31 (perhaps after poly-A clipping)
[2018-07-13 21:27:54.007] [jointLog] [warning] Entry with header [ENST00000450276.1|ENSG00000237020.1|OTTHUMG00000152432.2|OTTHUMT00000326210.2|AL122127.31-001|IGHD1-20|17|IG_D_gene|], had length less than the k-mer length of 31 (perhaps after poly-A clipping)
[2018-07-13 21:27:54.007] [jointLog] [warning] Entry with header [ENST00000390574.1|ENSG00000211914.1|OTTHUMG00000152431.2|OTTHUMT00000326209.2|AL122127.30-001|IGHD6-19|21|IG_D_gene|], had length less than the k-mer length of 31 (perhaps after poly-A clipping)
[2018-07-13 21:27:54.007] [jointLog] [warning] Entry with header [ENST00000390575.1|ENSG00000211915.1|OTTHUMG00000152433.2|OTTHUMT00000326211.2|AL122127.29-001|IGHD5-18|20|IG_D_gene|], had length less than the k-mer length of 31 (perhaps after poly-A clipping)
[2018-07-13 21:27:54.007] [jointLog] [warning] Entry with header [ENST00000431870.1|ENSG00000227800.1|OTTHUMG00000152437.2|OTTHUMT00000326215.2|AL122127.28-001|IGHD4-17|16|IG_D_gene|], had length less than the k-mer length of 31 (perhaps after poly-A clipping)
[2018-07-13 21:27:54.007] [jointLog] [warning] Entry with header [ENST00000451044.1|ENSG00000227108.1|OTTHUMG00000152369.2|OTTHUMT00000326003.2|AB019441.47-001|IGHD1-14|17|IG_D_gene|], had length less than the k-mer length of 31 (perhaps after poly-A clipping)
[2018-07-13 21:27:54.007] [jointLog] [warning] Entry with header [ENST00000390580.1|ENSG00000211920.1|OTTHUMG00000152370.2|OTTHUMT00000326004.2|AB019441.46-001|IGHD6-13|21|IG_D_gene|], had length less than the k-mer length of 31 (perhaps after poly-A clipping)
[2018-07-13 21:27:54.007] [jointLog] [warning] Entry with header [ENST00000390581.1|ENSG00000211921.1|OTTHUMG00000152367.2|OTTHUMT00000326001.2|AB019441.45-001|IGHD5-12|23|IG_D_gene|], had length less than the k-mer length of 31 (perhaps after poly-A clipping)
[2018-07-13 21:27:54.007] [jointLog] [warning] Entry with header [ENST00000431440.2|ENSG00000232543.2|OTTHUMG00000152368.2|OTTHUMT00000326002.2|AB019441.44-001|IGHD4-11|16|IG_D_gene|], had length less than the k-mer length of 31 (perhaps after poly-A clipping)
[2018-07-13 21:27:54.007] [jointLog] [warning] Entry with header [ENST00000430425.1|ENSG00000237197.1|OTTHUMG00000152357.2|OTTHUMT00000325963.2|AB019441.40-001|IGHD1-7|17|IG_D_gene|], had length less than the k-mer length of 31 (perhaps after poly-A clipping)
[2018-07-13 21:27:54.007] [jointLog] [warning] Entry with header [ENST00000454691.1|ENSG00000228131.1|OTTHUMG00000152353.2|OTTHUMT00000325959.2|AB019441.39-001|IGHD6-6|18|IG_D_gene|], had length less than the k-mer length of 31 (perhaps after poly-A clipping)
[2018-07-13 21:27:54.007] [jointLog] [warning] Entry with header [ENST00000390588.1|ENSG00000211928.1|OTTHUMG00000152360.2|OTTHUMT00000325966.2|AB019441.38-001|IGHD5-5|20|IG_D_gene|], had length less than the k-mer length of 31 (perhaps after poly-A clipping)
[2018-07-13 21:27:54.007] [jointLog] [warning] Entry with header [ENST00000414852.1|ENSG00000233655.1|OTTHUMG00000152355.2|OTTHUMT00000325961.2|AB019441.37-001|IGHD4-4|16|IG_D_gene|], had length less than the k-mer length of 31 (perhaps after poly-A clipping)
[2018-07-13 21:27:54.007] [jointLog] [warning] Entry with header [ENST00000454908.1|ENSG00000236170.1|OTTHUMG00000152359.2|OTTHUMT00000325965.2|AB019441.34-001|IGHD1-1|17|IG_D_gene|], had length less than the k-mer length of 31 (perhaps after poly-A clipping)
[2018-07-13 21:27:54.008] [jointLog] [warning] Entry with header [ENST00000518246.1|ENSG00000254045.1|OTTHUMG00000152060.1|OTTHUMT00000325154.1|AB019439.71-001|IGHVIII-22-2|28|IG_V_pseudogene|], had length less than the k-mer length of 31 (perhaps after poly-A clipping)
[2018-07-13 21:27:54.010] [jointLog] [warning] Entry with header [ENST00000604642.1|ENSG00000270961.1|OTTHUMG00000184622.2|OTTHUMT00000468982.2|RP11-1360M22.8-001|IGHD5OR15-5A|23|IG_D_gene|], had length less than the k-mer length of 31 (perhaps after poly-A clipping)
[2018-07-13 21:27:54.010] [jointLog] [warning] Entry with header [ENST00000603326.1|ENSG00000271317.1|OTTHUMG00000184621.3|OTTHUMT00000468981.3|RP11-1360M22.7-001|IGHD4OR15-4A|19|IG_D_gene|], had length less than the k-mer length of 31 (perhaps after poly-A clipping)
[2018-07-13 21:27:54.010] [jointLog] [warning] Entry with header [ENST00000605284.1|ENSG00000271336.1|OTTHUMG00000184580.2|OTTHUMT00000468908.2|RP11-1360M22.3-001|IGHD1OR15-1A|17|IG_D_gene|], had length less than the k-mer length of 31 (perhaps after poly-A clipping)
[2018-07-13 21:27:54.013] [jointLog] [warning] Entry with header [ENST00000604446.1|ENSG00000270824.1|OTTHUMG00000184624.2|OTTHUMT00000468984.2|RP11-810K23.15-001|IGHD5OR15-5B|23|IG_D_gene|], had length less than the k-mer length of 31 (perhaps after poly-A clipping)
[2018-07-13 21:27:54.013] [jointLog] [warning] Entry with header [ENST00000603693.1|ENSG00000270451.1|OTTHUMG00000184611.3|OTTHUMT00000468945.3|RP11-810K23.14-001|IGHD4OR15-4B|19|IG_D_gene|], had length less than the k-mer length of 31 (perhaps after poly-A clipping)
[2018-07-13 21:27:54.013] [jointLog] [warning] Entry with header [ENST00000604838.1|ENSG00000270185.1|OTTHUMG00000184585.2|OTTHUMT00000468915.2|RP11-1360M22.4-001|IGHD1OR15-1B|17|IG_D_gene|], had length less than the k-mer length of 31 (perhaps after poly-A clipping)
[2018-07-13 21:27:55.348] [jointLog] [warning] Entry with header [ENST00000579054.1|ENSG00000266416.1|OTTHUMG00000179204.1|OTTHUMT00000445280.1|RP1-66C13.2-001|RP1-66C13.2|28|processed_pseudogene|], had length less than the k-mer length of 31 (perhaps after poly-A clipping)
[2018-07-13 21:27:56.453] [jointLog] [warning] Entry with header [ENST00000634174.1|ENSG00000282732.1|OTTHUMG00000191398.1|OTTHUMT00000487783.1|RP11-157B13.10-001|RP11-157B13.10|28|unprocessed_pseudogene|], had length less than the k-mer length of 31 (perhaps after poly-A clipping)
Elapsed time: 20.6564s

[2018-07-13 21:27:58.139] [jointLog] [warning] Removed 808 transcripts that were sequence duplicates of indexed transcripts.
[2018-07-13 21:27:58.139] [jointLog] [warning] If you wish to retain duplicate transcripts, please use the `--keepDuplicates` flag
Replaced 4 non-ATCG nucleotides
Clipped poly-A tails from 1586 transcripts
Building rank-select dictionary and saving to disk done
Elapsed time: 0.0133569s
Writing sequence data to file . . . done
Elapsed time: 0.182628s
[info] Building 32-bit suffix array (length of generalized text is 308972089)
Building suffix array . . . success
saving to disk . . . done
Elapsed time: 0.513756s
done
Elapsed time: 58.2945s
processed 308000000 positions
khash had 130317526 keys
saving hash to disk . . . done
Elapsed time: 9.6539s
[2018-07-13 21:32:25.842] [jLog] [info] done building index
(salmon)

**Comment by sunnyyakima** [comment]
the new json files from the latest index build are not empty. and 
salmon quant is running now, so far no error!
Thanks for the quick response!

**Comment by rob-p** [comment]
No worries.  Not sure what happened in the first index run, but if it happens again, please feel free to re-open this issue or to open a new one and reference this one.

Best,
Rob

**Comment by FerJmnzCc** [comment]
Hi, I have some kind the same error. I download the prebuild index from refgenie and I got exactly the same error message. 

refgenie pull hg38/salmon_sa_index <- I downloaded the 16Gb of the index files

[2020-05-04 21:30:58.648] [jointLog] [info] Fragment incompatibility prior below threshold.  Incompatible fragments will be ignored.
[2020-05-04 21:30:58.648] [jointLog] [info] Usage of --validateMappings implies use of minScoreFraction. Since not explicitly specified, it is being set to 0.65
[2020-05-04 21:30:58.648] [jointLog] [info] Usage of --validateMappings, without --hardFilter implies use of range factorization. rangeFactorizationBins is being set to 4
[2020-05-04 21:30:58.648] [jointLog] [info] Usage of --validateMappings implies a default consensus slack of 0.2. Setting consensusSlack to 0.2.
[2020-05-04 21:30:58.648] [jointLog] [info] parsing read library format
[2020-05-04 21:30:58.648] [jointLog] [info] There is 1 library.
[2020-05-04 21:30:58.701] [jointLog] [info] Loading Quasi index
Exception : [rapidjson internal assertion failure: IsObject()]
salmon quant was invoked improperly.
For usage information, try salmon quant --help
Exiting.

The son files of the index show this;
ls -lrth *json
-rwxrwxrwx 1 usr usr 1007 dic 14 00:41 info.json
-rwxrwxrwx 1 usr usr   96 dic 14 00:44 versionInfo.json

Any idea would be really appreciated,

Kind regards, 
Fer
